tests/mock_crust/mod.rs

```diff
@@ -54,6 +75,51 @@ fn disconnect_on_rebootstrap() {
     expect_next_event!(unwrap!(nodes.last_mut()), Event::Terminated);
 }
 
+#[test]
+fn candidate_timeout_resource_proof() {
+    let network = Network::new(MIN_SECTION_SIZE, None);
+    let mut nodes = create_connected_nodes_until_split(&network, vec![1, 1], false);
+    let bootstrap_config = BootstrapConfig::with_contacts(&[nodes[0].handle.endpoint()]);
+    nodes.insert(
+        0,
+        TestNode::builder(&network)
+            .bootstrap_config(bootstrap_config)
+            .create(),
+    );
+
+    // Initiate connection until the node receive RelocateResponse:
+    info!("Candidate joining name: {}", nodes[0].name());
+    poll_and_resend_until(&mut nodes, &mut [], &|nodes| {
+        nodes[0].inner.is_proving_node()
+    });
+    let proving_node = nodes.remove(0);
+
+    assert!(
+        proving_node.inner.is_proving_node(),
+        "Accepted as candidate"
+    );
+
+    // Continue without the joining node until all node consensused events:
+    info!("Candidate new name: {}", proving_node.name());
+    poll_and_resend(&mut nodes, &mut []);
+
+    assert_eq!(
+        nodes_in_authority(&nodes, &proving_node.name()),
+        nodes_with_candidate(&nodes),
+        "All members of destination section accepted node as candidate"
+    );
+
+    // Continue after candidate time out:
+    FakeClock::advance_time(60 * 60 * 1000);
```

Comment:
ah I just put this in my diff as a tmp placeholder. not that it should affect anything but if wanting to be precise, we could advance time specifically using the RES_PROOF consts to forward time and check accordingly.

Reply:
I think I'll leave it as is.
Because node could (will?) have different starting instant, it seem difficult to try to be precise here.